Added distributed lock api support to javascript sdk (#306)
dapr/js-sdk: examples/distributedLock/README.md

# Example - Distributed Lock APIs

> This example demonstrates the Distributed Lock APIs in Dapr.

Reviewer: Maybe rework this to:

This example demonstrates 2 features (Try Lock & Unlock) from the [Distributed lock API from Dapr](https://github.com/dapr/dapr/issues/3549) that helps developers to keep their data safe from race conditions. For more information, check out the documentation at <insert link>

It demonstrates the following APIs:
- **Distributed Lock**: Try Lock
- **Distributed Lock**: Unlock

> **Note:** Make sure to use the latest proto bindings by running scripts/fetch-proto.sh file.
## Prerequisites

- [Dapr CLI](https://docs.dapr.io/getting-started/install-dapr-cli/)
- [Dapr JS SDK](https://docs.dapr.io/developing-applications/sdks/js/)
- Initialize Dapr environment by pulling the code from master branch of [Dapr](https://github.com/dapr/dapr)

## Overview

> This example shows the usage of two different Distributed Lock APIs, TryLock call and Unlock call.

Reviewer: Redundant to the intro? Maybe write: "The TryLock and Unlock calls are implemented under the `client.lock` attribute" ?

#### TryLock Example
```javascript
const tryLockResponse = await client.distributedLock.tryLock(storeName, resourceId, lockOwner, expiryInSeconds);
```

#### Unlock Example
```javascript
const unLockResponse = await client.distributedLock.unlock(storeName, resourceId, lockOwner);
```

> ### Start the DistributedLock TryLockApplication.

Reviewer: `DistributedLock` to `Lock`

### Start the Lock application

> Change directory to this folder:

Reviewer: Execute the example under the folder `examples/Client/LockAPI/TryLock`

```bash
cd examples/Client/DistributedLockApi/TryLockApplication
npm install
```

> To run the `TryLockApplication`, execute the following command:

Reviewer: the `TryLock`

```bash

> dapr run --app-id distributedLock --app-protocol grpc --components-path ./components npm run start

Reviewer: `--app-id lock`

```

You should see the following output from the application:

```
== APP == Acquiring lock on redislock, resourceId as owner: owner1
== APP == { success: true }
== APP == Unlocking on redislock, resourceId as owner: owner1
== APP == Unlock API response: Success
== APP == Unlocking on redislock, resourceId as owner: owner1
== APP == Unlock API response when lock is not acquired: LockUnexist
== APP == Acquiring lock on redislock, resourceId as owner: owner1
== APP == Acquired Lock? true
```

> ### Start the DistributedLock UnLockApplication.

Reviewer: DistributedLock to Lock

### Start the Lock UnLock Example

Run `UnLockApplication` after `TryLockApplication` is ran.

> Change directory to this folder:

Reviewer: Navigate to `examples/Client/LockAPI/Unlock`

```bash
cd examples/Client/DistributedLockApi/UnLockApplication
npm install
```

To run the `UnLockApplication`, execute the following command:

```bash

> dapr run --app-id distributedLock --app-protocol grpc --components-path ./components npm run start

Reviewer: `--app-id lock`

> # Example - Distributed Lock APIs

Reviewer 2: Shall we create a separate issue for adding this to SDK docs? Or should we update the same in this PR?

Author: I would keep this in the same PR.

Reviewer 2: This is example, but we still need SDK doc, right?

This example demonstrates 2 features (Try Lock & Unlock) from the [Distributed lock API from Dapr](https://github.com/dapr/dapr/issues/3549) that helps developers to keep their data safe from race conditions. For more information, check out the documentation at 

It demonstrates the following APIs:
- **Distributed Lock**: Try Lock
- **Distributed Lock**: Unlock

> **Note:** Make sure to use the latest proto bindings by running scripts/fetch-proto.sh file.
## Prerequisites

- [Dapr CLI](https://docs.dapr.io/getting-started/install-dapr-cli/)
- [Dapr JS SDK](https://docs.dapr.io/developing-applications/sdks/js/)
- Initialize Dapr environment by pulling the code from master branch of [Dapr](https://github.com/dapr/dapr)

## Overview

> The TryLock and Unlock calls are implemented under the client.lock attribute.

Reviewer: nit: newline for Markdown

#### TryLock Example

> ```javascript

Reviewer: nit: newline for Markdown

const tryLockResponse = await client.distributedLock.tryLock(storeName, resourceId, lockOwner, expiryInSeconds);
```

#### Unlock Example

> ```javascript

Reviewer: nit: newline for Markdown

> const unLockResponse = await client.distributedLock.unlock(storeName, resourceId, lockOwner);

Reviewer: unLock -> unlock (it should be 1 word)

```

### Start the Lock application.

> Execute the example under the folder examples/Client/LockAPI/TryLock.

Reviewer: wrap in \` -> \`examples/Client/LockAPI/TryLock\`

```bash
cd examples/Client/DistributedLockApi/TryLockApplication
npm install
```

To run the `TryLock`, execute the following command:

```bash
dapr run --app-id lock --app-protocol grpc --components-path ./components npm run start
```

You should see the following output from the application:

```
== APP == Acquiring lock on redislock, resourceId as owner: owner1
== APP == { success: true }
== APP == Unlocking on redislock, resourceId as owner: owner1
== APP == Unlock API response: Success
== APP == Unlocking on redislock, resourceId as owner: owner1
== APP == Unlock API response when lock is not acquired: LockUnexist
== APP == Acquiring lock on redislock, resourceId as owner: owner1
== APP == Acquired Lock? true
```

> ### Start the Lock UnLock Example.

Reviewer: Stay consistent in titles, first we did "Start the Lock Application" so now I would do "Start the Unlock Application"

> Run `UnLockApplication` after `TryLockApplication` is ran.

Reviewer: UnLock -> unlock (it should be 1 word)

Navigate to examples/Client/LockAPI/Unlock.

```bash

> cd examples/Client/DistributedLockApi/UnLockApplication

Reviewer: UnLock -> unlock (it should be 1 word)

npm install
```

> To run the `UnLockApplication`, execute the following command:

Reviewer: UnLock -> unlock (it should be 1 word)

# Example - Distributed Lock APIs

This example demonstrates 2 features (Try Lock & Unlock) from the [Distributed lock API from Dapr](https://github.com/dapr/dapr/issues/3549) that helps developers to keep their data safe from race conditions. For more information, check out the documentation at 

> It demonstrates the following APIs:
> - **Distributed Lock**: Try Lock
> - **Distributed Lock**: Unlock

Reviewer 2: ```suggestion
It demonstrates the **Distributed Lock** API's following methods:
- `TryLock`
- `Unlock`
```

> **Note:** Make sure to use the latest proto bindings by running scripts/fetch-proto.sh file.
## Prerequisites

- [Dapr CLI](https://docs.dapr.io/getting-started/install-dapr-cli/)
- [Dapr JS SDK](https://docs.dapr.io/developing-applications/sdks/js/)
- Initialize Dapr environment by pulling the code from master branch of [Dapr](https://github.com/dapr/dapr)

## Overview

The TryLock and Unlock calls are implemented under the client.lock attribute.

#### TryLock Example

```javascript

> const tryLockResponse = await client.distributedLock.tryLock(storeName, resourceId, lockOwner, expiryInSeconds);

Reviewer 2: ```suggestion
const tryLockResponse = await client.lock.tryLock(storeName, resourceId, lockOwner, expiryInSeconds);
```
(Other places too)

```

#### Unlock Example

```javascript
const unlockResponse = await client.distributedLock.unlock(storeName, resourceId, lockOwner);
```

### Start the Lock application.

Execute the example under the folder `examples/Client/LockAPI/TryLock`

```bash
cd examples/Client/DistributedLockApi/TryLockApplication
npm install
```

To run the `TryLock`, execute the following command:

```bash
dapr run --app-id lock --app-protocol grpc --components-path ./components npm run start
```

You should see the following output from the application:

```
== APP == Acquiring lock on redislock, resourceId as owner: owner1
== APP == { success: true }
== APP == Unlocking on redislock, resourceId as owner: owner1
== APP == Unlock API response: Success
== APP == Unlocking on redislock, resourceId as owner: owner1
== APP == Unlock API response when lock is not acquired: LockDoesNotExist
== APP == Acquiring lock on redislock, resourceId as owner: owner1
== APP == Acquired Lock? true
```

> ### Start the UnLock Example.

Reviewer 2: ```suggestion
### Start the Unlock Example.
```

Run `UnlockApplication` after `TryLockApplication` is ran.

Navigate to examples/Client/LockAPI/Unlock.

```bash

> cd examples/Client/DistributedLockApi/UnlockApplication

Reviewer 2: Is this path correct?
